fem/linearform.cpp

```diff
@@ -92,14 +114,31 @@ void LinearForm::Assemble()
 
    if (dlfi.Size())
    {
+      for (int k = 0; k < dlfi.Size(); k++)
+      {
+         if (dlfi_elem_attr_marker[k] != NULL)
+         {
+            MFEM_VERIFY(fes->GetMesh()->attributes.Size() ==
+                        dlfi_elem_attr_marker[k]->Size(),
+                        "invalid element marker for domain linear form "
+                        "integrator #" << k << ", counting from zero");
+         }
+      }
+
       for (i = 0; i < fes -> GetNE(); i++)
       {
+         int elem_attr = fes->GetMesh()->GetAttribute(i);
          fes -> GetElementVDofs (i, vdofs);
```

Comment:
Same as in `bilinearform`:
It would be good to skip all computations (`GetElementDofs`, `GetElementTransformation` etc) if the element's attribute is not active for any integrator. E.g., see the logic with `bdr_attr_marker` below.